hawkbit-repository/hawkbit-repository-jpa/src/test/java/org/eclipse/hawkbit/repository/jpa/RolloutManagementTest.java

```diff
@@ -161,6 +163,105 @@ public void checkRunningRolloutsDoesNotStartNextGroupIfFinishConditionIsNotHit()
+
+        checkStatusOfRolloutAndRolloutGroupsAfterDeletingTargets(createdRollout);
+
+    }
+
+    @Step("Create a rollout by the given parameter and start it.")
+    public Rollout createAndStartRollout(final int amountTargetsForRollout, final int amountOtherTargets,
+            final int amountGroups, final String successCondition, final String errorCondition) {
+
+        final Rollout createdRollout = createSimpleTestRolloutWithTargetsAndDistributionSet(amountTargetsForRollout,
+                amountOtherTargets, amountGroups, successCondition, errorCondition);
+        rolloutManagement.startRollout(createdRollout);
+        return createdRollout;
+    }
+
+    @Step("Finish three actions of the rollout group and delete two targets")
+    public void finishActionAndDeleteTargetsOfFirstRunningGroup(final Rollout createdRollout) {
+        // finish group one by finishing targets and deleting targets
+        final List<Action> runningActions = deploymentManagement.findActionsByRolloutAndStatus(createdRollout,
+                Status.RUNNING);
+        finishAction(runningActions.get(0));
+        finishAction(runningActions.get(1));
+        finishAction(runningActions.get(2));
+        targetManagement.deleteTargets(runningActions.get(3).getTarget().getId(),
+                runningActions.get(4).getTarget().getId());
+    }
+
+    @Step("Check the status of the rollout groups")
+    public void checkStatusOfRolloutGroupsAfterDeletingTargets(final Rollout createdRollout) {
+        rolloutManagement.checkRunningRollouts(0);
+        // validate that the second group is in running state
+        final List<RolloutGroup> runningRolloutGroups = rolloutGroupManagement.findRolloutGroupsByRolloutId(
+                createdRollout.getId(), new OffsetBasedPageRequest(0, 10, new Sort(Direction.ASC, "id"))).getContent();
+        assertThat(runningRolloutGroups.get(0).getStatus()).isEqualTo(RolloutGroupStatus.FINISHED);
+        assertThat(runningRolloutGroups.get(1).getStatus()).isEqualTo(RolloutGroupStatus.RUNNING);
+        assertThat(runningRolloutGroups.get(2).getStatus()).isEqualTo(RolloutGroupStatus.SCHEDULED);
+    }
+
+    @Step("Finish one action of the rollout group and delete four targets")
+    public void finishActionAndDeleteTargetsOfSecondRunningGroup(final Rollout createdRollout) {
+        final List<Action> runningActions = deploymentManagement.findActionsByRolloutAndStatus(createdRollout,
+                Status.RUNNING);
+        finishAction(runningActions.get(0));
+        targetManagement.deleteTargets(runningActions.get(1).getTarget().getId(),
+                runningActions.get(2).getTarget().getId(), runningActions.get(3).getTarget().getId(),
+                runningActions.get(4).getTarget().getId());
+
+    }
+
+    @Step("Delete all targets of the rollout group")
+    public void deleteThrirdGroup(final Rollout createdRollout) {
+        // delete all targets of the third group
```

Comment:
Rename method to `deleteAllTargetsFromThirdGroup(...)`, than you don't need the comment.